build error on vcursor_impl tests on newer versions of go

When filing a bug, please include the following headings if
possible. Any example text in this template can be deleted.

#### Overview of the Issue

When running the `go/vt/vtgate/vcursor_impl_test.go` tests, I get the following on go1.15

```
Running tool: /usr/local/bin/go test -timeout 30s -coverprofile=/var/folders/38/3_q_szqn7zx66d0cgj0g_z1w0000gp/T/vscode-go0YkAyW/go-code-cover vitess.io/vitess/go/vt/vtgate -run ^(TestDestinationKeyspace|TestSetTarget|TestPlanPrefixKey|TestFirstSortedKeyspace)$

# vitess.io/vitess/go/vt/vtgate
/Users/amason/work/vitess/go/vt/vtgate/vcursor_impl_test.go:240:9: conversion from int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
/Users/amason/work/vitess/go/vt/vtgate/vcursor_impl_test.go:280:9: conversion from int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
FAIL	vitess.io/vitess/go/vt/vtgate [build failed]
FAIL
```

This also happens if I run `go test -v ./go/vt/vtgate` myself.

#### Reproduction Steps

Install go1.15 and go1.14:

```
❯ /usr/local/opt/go@1.15/bin/go test -v ./go/vt/vtgate
# vitess.io/vitess/go/vt/vtgate
go/vt/vtgate/vcursor_impl_test.go:240:9: conversion from int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
go/vt/vtgate/vcursor_impl_test.go:280:9: conversion from int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
FAIL    vitess.io/vitess/go/vt/vtgate [build failed]
FAIL
❯ /usr/local/opt/go@1.14/bin/go test -v ./go/vt/vtgate
ERROR: logging before flag.Parse: I0929 19:04:38.557277   31538 buffer.go:144] vtgate buffer not enabled.
ERROR: logging before flag.Parse: I0929 19:04:38.557461   31538 discoverygateway.go:122] loading tablets for cells: 
ERROR: logging before flag.Parse: I0929 19:04:38.557488   31538 gateway.go:100] Gateway waiting for serving tablets of types [] ...
ERROR: logging before flag.Parse: I0929 19:04:38.557512   31538 gateway.go:108] Waiting for tablets completed
ERROR: logging before flag.Parse: I0929 19:04:38.557523   31538 vtgate.go:85] Transaction mode: 'MULTI'
ERROR: logging before flag.Parse: I0929 19:04:38.558026   31538 vschema_manager.go:74] Received vschema update
ERROR: logging before flag.Parse: I0929 19:04:38.558196   31538 streamlog.go:155] Streaming logs from VTGate at /debug/querylog.
=== RUN   TestAutocommitUpdateSharded
--- PASS: TestAutocommitUpdateSharded (0.00s)
[rest omitted]
```

--- a/go/vt/vtgate/vcursor_impl_test.go
+++ b/go/vt/vtgate/vcursor_impl_test.go
@@ -3,6 +3,7 @@ package vtgate
 import (
 	"context"
 	"encoding/hex"
+	"fmt"
 	"testing"
 
 	"vitess.io/vitess/go/vt/proto/vschema"
@@ -237,7 +238,7 @@ func TestSetTarget(t *testing.T) {
 	}}
 
 	for i, tc := range tests {
-		t.Run(string(i)+"#"+tc.targetString, func(t *testing.T) {
+		t.Run(fmt.Sprintf("%d#%s", i, tc.targetString), func(t *testing.T) {
 			vc, _ := newVCursorImpl(context.Background(), NewSafeSession(&vtgatepb.Session{InTransaction: true}), sqlparser.MarginComments{}, nil, nil, &fakeVSchemaOperator{vschema: tc.vschema}, tc.vschema, nil)
 			vc.vschema = tc.vschema
 			err := vc.SetTarget(tc.targetString)
@@ -277,7 +278,7 @@ func TestPlanPrefixKey(t *testing.T) {
 	}}
 
 	for i, tc := range tests {
-		t.Run(string(i)+"#"+tc.targetString, func(t *testing.T) {
+		t.Run(fmt.Sprintf("%d#%s", i, tc.targetString), func(t *testing.T) {
 			ss := NewSafeSession(&vtgatepb.Session{InTransaction: false})
 			ss.SetTargetString(tc.targetString)
 			vc, err := newVCursorImpl(context.Background(), ss, sqlparser.MarginComments{}, nil, nil, &fakeVSchemaOperator{vschema: tc.vschema}, tc.vschema, srvtopo.NewResolver(&fakeTopoServer{}, nil, ""))

